RealmTasks Android/app/src/main/java/io/realm/realmtasks/list/ItemViewHolder.java

```diff
@@ -55,34 +69,49 @@ public ItemViewHolder(View itemView, RecyclerView.Adapter adapter) {
         badge = (TextView) row.findViewById(R.id.badge);
         text = (TextView) row.findViewById(R.id.text);
         editText = (EditText) row.findViewById(R.id.edit_text);
+        cellUnusedColor = ContextCompat.getColor(itemView.getContext(), R.color.cell_unused_color);
+        cellCompletedColor = ContextCompat.getColor(itemView.getContext(), R.color.cell_completed_color);
+        cellCompletedBackgroundColor = ContextCompat.getColor(itemView.getContext(), R.color.cell_completed_background_color);
+        cellDefaultColor = ContextCompat.getColor(itemView.getContext(), R.color.cell_default_color);
+        shouldChangeBackgroundColor = true;
+        previousFirstLength = -1;
         this.adapter = adapter;
     }
 
     private int generateBackgroundColor() {
         if (adapter != null && adapter instanceof TouchHelperAdapter) {
             return ((TouchHelperAdapter) adapter).generatedRowColor(getAdapterPosition());
         } else {
-            return UNUSED_COLOR;
+            return cellUnusedColor;
         }
     }
 
     public void setCompleted(boolean completed) {
+        if (completed == this.completed) {
+            return;
```

Comment:
Thanks @zaki50 You're right. It cannot be skipped by `completed == this.completed`. I should modify this part.